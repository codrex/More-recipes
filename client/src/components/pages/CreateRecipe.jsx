import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import RecipeEditor from './RecipeEditor';
import { resetSuccess } from '../../actions/ajaxActions';
import { RECIPE_ADDED } from '../../constants';
import {
  createRecipe,
  getRecipe,
  resetRecipe
} from '../../actions/recipeActions';

/**
 * CreateRecipe
 */
class CreateRecipe extends React.Component {
  /**
   * constructor
   * @return {undefined}
   * @param {Object} props
   */
  constructor(props) {
    super(props);
    props.resetRecipe();
  }

  /**
   * @returns {undefined}
   * @param {object} nextProps
   */
  componentWillReceiveProps(nextProps) {
    const {
      message,
      history,
      resetSuccessMessage
    } = nextProps;

    if (message === RECIPE_ADDED) {
      resetSuccessMessage();
      history.push(`/recipe/${nextProps.recipe.id}`);
    }
  }

  /**
   * post recipe button click event handler
   * @return {undefined}
   */
  postRecipeClicked = () => {
    const { recipe } = this.props;
    this.props.createRecipe(recipe, RECIPE_ADDED);
  }

  /**
   * render
   * @return {React} - react component
   */
  render() {
    const { loading } = this.props;

    return (
      <RecipeEditor
        handleButtonClick={this.postRecipeClicked}
        submitButtonText="post recipe"
        loading={loading}
        hasRender
      />
    );
  }
}

CreateRecipe.propTypes = {
  recipe: PropTypes.objectOf(PropTypes.shape).isRequired,
  history: PropTypes.objectOf(PropTypes.shape).isRequired,
  resetRecipe: PropTypes.func.isRequired,
  createRecipe: PropTypes.func.isRequired,
  resetSuccessMessage: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  message: PropTypes.string,
};

CreateRecipe.defaultProps = {
  message: '',
};

const mapStateToProps = state => (
  {
    recipe: state.recipe,
    loading: state.networkRequest.loading,
    message: state.networkRequest.msg,
    statusCode: state.currentStatusCode
  }
);

export { CreateRecipe as PureCreateRecipe };
export default connect(mapStateToProps, {
  resetRecipe,
  createRecipe,
  getRecipe,
  resetSuccessMessage: resetSuccess
})(CreateRecipe);
