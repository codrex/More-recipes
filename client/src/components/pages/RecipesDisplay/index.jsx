import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import RecipeGrid from './RecipeGrid';
import Paginator from '../../common/Paginator';
import NotFound from '../../common/NotFound';
import HeroArea from '../../common/HeroArea';
import { DEFAULT_PIX } from '../../../constants';
import { toggleFav } from '../../../actions/recipeActions';
import resetPageCount from '../../../actions/resetPageCount';
import { resetSuccess } from '../../../actions/ajaxActions';

/**
 * RecipesDisplay
 */
class RecipesDisplay extends React.Component {
  state ={
    recipes: this.props.recipes
  }

  /**
   * @return {undefined}
   */
  componentDidMount() {
    this.props.resetPageCount();
    this.props.getRecipes(1);
  }

  /**
   * @return {undefined}
   * @param {object} nextProps
   */
  componentWillReceiveProps(nextProps) {
    if (this.state.recipes === nextProps.recipes) return;
    this.setState({ recipes: nextProps.recipes });
  }

  /**
   * @return {undefined}
   */
  componentWillUnmount() {
    this.props.resetSuccess();
  }

  /**
   * check if recipe id is in the list of user's fav recipes
   * @return {undefined}
   * @param {number} id
   */
  isUserFav = id => this.props.user.favRecipes.find((recipe) => {
    const recipeId = recipe.id;
    return id === recipeId;
  })

  /**
   * toggle favourite state
   * @return {undefined}
   * @param {number} id
   */
  toggleFav = (id) => {
    this.props.resetSuccess();
    const message = this.isUserFav(id) ? 'Recipe removed from favourites' :
      'Recipe added to favourites';
    this.props.toggleFav(id, message);
  }

  /**
   * Recipe Grid
   * @return {React} react component
   */
  renderRecipeGrid = () => {
    const { loading, history } = this.props;
    return (
      <RecipeGrid
        recipes={this.state.recipes}
        loading={loading}
        toggleFav={this.toggleFav}
        isUserFav={this.isUserFav}
        history={history}
      />
    );
  }

  /**
   * Pagination
   * @return {React} react component
   */
  renderPagination = () => {
    const {
      pageCount,
      getRecipes,
      loading
    } = this.props;

    if (pageCount > 1) {
      return (
        <Paginator
          loading={loading}
          pageCount={pageCount}
          handlePageClick={({ selected }) => {
            getRecipes(selected + 1);
          }}
        />
      );
    } return null;
  }

  /**
   * NotFound
   * @return {React} react component
   */
  renderNotFound = () => {
    const {
      loading,
      recipes
    } = this.props;

    if (recipes.length < 1 && !loading) {
      return (
        <NotFound message="No recipe found" />
      );
    } return null;
  }

  /**
   * render
   * @return {React} react component
   */
  render() {
    const { loading, recipes, title } = this.props;
    return (
      <div className="container-fluid no-padding" id="recipes">
        <HeroArea image={DEFAULT_PIX} title={title} />
        <div className="row recipes-board d-flex flex-column">
          {!loading && recipes.length > 0 &&
            <div
              id="display-1"
              className={
                `bold
                text-dark
                d-flex
                justify-content-between`
              }
            >
              <span className="lead">
                {`Total Recipes: ${recipes.length}`}
              </span>

            </div>
          }
          {this.renderRecipeGrid()}
          {this.renderPagination()}
          {this.renderNotFound()}
        </div>
      </div>
    );
  }
}

RecipesDisplay.propTypes = {
  loading: PropTypes.bool.isRequired,
  recipes: PropTypes.arrayOf(PropTypes.shape).isRequired,
  user: PropTypes.objectOf(PropTypes.shape).isRequired,
  history: PropTypes.objectOf(PropTypes.shape).isRequired,
  pageCount: PropTypes.number,
  title: PropTypes.string,
  resetPageCount: PropTypes.func.isRequired,
  resetSuccess: PropTypes.func.isRequired,
  getRecipes: PropTypes.func.isRequired,
  toggleFav: PropTypes.func.isRequired
};

RecipesDisplay.defaultProps = {
  pageCount: 0,
  title: ''
};

const mapStateToProps = state => ({
  user: state.user,
  loading: state.networkRequest.loading,
  requestCount: state.networkRequest.requestCount,
  auth: state.auth,
  pageCount: state.pageCount,
});

export { RecipesDisplay as PureRecipesDisplay };

export default connect(mapStateToProps, {
  toggleFav,
  resetPageCount,
  resetSuccess
})(RecipesDisplay);
