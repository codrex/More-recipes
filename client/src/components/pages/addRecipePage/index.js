import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import Icon from '../../common/icon/icon';
import AddDirections from './addDirections/addDirections';
import AddIngredients from './addIngredients/addIngredients';
import RecipeNameAndCategory from './recipeNameAndCategory/recipeNameAndCategory';
import { createRecipe, updateAllRecipeField } from '../../../actions/recipeActions';
import { ajaxRedirect } from '../../../actions/ajaxActions';
import './addRecipe.scss';

/**
 * Add recipe component
 */
class AddRecipes extends React.Component {

  /**
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.saveRecipe = this.saveRecipe.bind(this);
  }
/**
 *
 * @param {object} nextProps
 * @return {bool} true or false
 */
  shouldComponentUpdate(nextProps) {
    if (nextProps.redirectUrl === '/') {
      this.props.actions.redirect('/recipe/create');
      this.props.history.push(nextProps.redirectUrl);
      return false;
    }
    if (nextProps.recipe !== this.props.recipe) {
      this.props.actions.updateLocalRecipe(nextProps.recipe);
      return true;
    }
    return true;
  }

  /**
   * @return {undefined}
   * @param {Object} values
   */
  saveRecipe(values) {
    this.props.actions.updateStore(values);
  }
  /**
   * @return {undefined}
   */
  render() {
    return (
      <div className="container-fluid no-padding ">
        <div className="row d-flex">
          <div className="col-xs-9 col-sm-9 col-md-10 col-lg-4 name-category no-padding">
            <h4 className="lead items-header-text">
              Add recipe
              <Icon
                iconClass="fa fa-plus"
                className="float-right"
                handleClick={this.newItem}
              />

            </h4>
            <div className="col-12">
              <RecipeNameAndCategory createRecipe={createRecipe} />
            </div>
          </div>
          <div className="col-xs-9 col-sm-9 col-md-10 col-lg-4 d-flex ingredients no-padding">
            <AddIngredients />
          </div>
          <div className="col-xs-9 col-sm-9 col-md-10 col-lg-4 d-flex directions no-padding">
            <AddDirections />
          </div>
        </div>
      </div>
    );
  }
}

AddRecipes.propTypes = {
  actions: PropTypes.object,
  loading: PropTypes.bool,
  redirectUrl: PropTypes.string,
  history: PropTypes.object,
  redirect: PropTypes.func,
  recipe: PropTypes.object,

};

const mapStateToProps = (state) => (
  {
    loading: state.ajaxCall > 0,
    redirectUrl: state.redirectUrl,
    recipe: state.recipe,
  }
);
const mapDispatchToProps = (dispatch) => (
  {
    actions: {
      redirect: bindActionCreators(ajaxRedirect, dispatch),
      updateLocalRecipe: bindActionCreators(updateAllRecipeField, dispatch)
    }
  }
);

export default connect(mapStateToProps, mapDispatchToProps)(AddRecipes);
