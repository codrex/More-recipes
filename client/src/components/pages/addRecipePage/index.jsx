import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import TopBar from '../../common/topbar/topbar';
import AddDirections from './addDirections/addDirections';
import AddIngredients from './addIngredients/addIngredients';
import RecipeNameAndCategory from './recipeNameAndCategory/recipeNameAndCategory';
import {
  updateAllRecipeField,
  recipeToModify,
  getRecipe,
  onNewRecipe
} from '../../../actions/recipeActions';
import { ajaxRedirect } from '../../../actions/ajaxActions';

/**
 * Add recipe component
 */
class AddRecipes extends React.Component {
  /**
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.create = props.match.path === '/recipe/create';
    this.CREATE_RECIPE_SUCCESS = 'Recipe added successfully';
    this.MODIFY_RECIPE_SUCCESS = 'Recipe modified successfully';
  }

  /**
 * @return {undefined}
 */
  componentDidMount() {
    if (!this.create) {
      const id = parseInt(this.props.match.params.id, 10);
      if (this.props.recipes) {
        const recipe = this.props.recipes.find(element => element.id === id);
        this.props.actions.recipeToModify(recipe);
      } else {
        this.props.actions.getRecipe(id);
      }
    } else {
      this.props.actions.newRecipe();
    }
  }
  /**
   * @return {undefined}
   * @param {Object} nextProps
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.success.success === this.CREATE_RECIPE_SUCCESS) {
      this.props.actions.newRecipe();
      this.props.actions.clearSuccessMsg({
        success: ''
      });
    }
  }

  /**
 *
 * @param {object} nextProps
 * @return {bool} true or false
 */
  shouldComponentUpdate(nextProps) {
    if (nextProps.redirectUrl === '/') {
      this.props.actions.redirect('/recipe/create');
      this.props.history.push(nextProps.redirectUrl);
      return false;
    }
    if (nextProps.recipe !== this.props.recipe) {
      this.props.actions.updateLocalRecipe(nextProps.recipe);
      return true;
    }
    return true;
  }

  /**
   * @return {undefined}
   */
  render() {
    return (
      <div
        className="container"
        style={{
          maxWidth: 'unset'
        }}
      >
        <div className="row">
          <TopBar title={this.create ? 'Add recipe' : 'Modify recipe'} />
        </div>
        <div
          className="row d-flex justify-content-between align-items-start"
          style={{
            margin: '0.5em'
          }}
        >
          <div className="col-xs-11 col-sm-11 col-md-10 col-lg-4 name-category no-padding">
            <div className="col-12">
              <RecipeNameAndCategory
                postRecipe={this.create}
                loading={this.props.loading}
                create={this.CREATE_RECIPE_SUCCESS}
                modify={this.MODIFY_RECIPE_SUCCESS}
              />
            </div>
          </div>
          <div className="col-xs-11 col-sm-11 col-md-10 col-lg-4 d-flex ingredients no-padding">
            <AddIngredients />
          </div>
          <div className="col-xs-11 col-sm-11 col-md-10 col-lg-4 d-flex directions no-padding">
            <AddDirections />
          </div>
        </div>
      </div>
    );
  }
}

AddRecipes.propTypes = {
  actions: PropTypes.object,
  loading: PropTypes.bool,
  redirectUrl: PropTypes.string,
  history: PropTypes.object,
  redirect: PropTypes.func,
  recipe: PropTypes.object,
  recipes: PropTypes.array,
  match: PropTypes.object
};

const mapStateToProps = state => ({
  loading: state.ajaxCall > 0,
  redirectUrl: state.redirectUrl,
  // recipes: state.user.createdRecipes,
  success: state.ajaxSuccess
});
const mapDispatchToProps = dispatch => ({
  actions: {
    redirect: bindActionCreators(ajaxRedirect, dispatch),
    updateLocalRecipe: bindActionCreators(updateAllRecipeField, dispatch),
    recipeToModify: bindActionCreators(recipeToModify, dispatch),
    getRecipe: bindActionCreators(getRecipe, dispatch),
    newRecipe: bindActionCreators(onNewRecipe, dispatch),
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(AddRecipes);
