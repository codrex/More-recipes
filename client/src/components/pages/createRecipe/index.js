import CreateRecipe from './createRecipe';

export default CreateRecipe;
