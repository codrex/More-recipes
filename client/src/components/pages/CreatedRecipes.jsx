import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getCreatedRecipes } from '../../actions/recipeActions';
import RecipesDisplay from './RecipesDisplay';

/**
 * CreateRecipes component
 * @param {object} props
 * @return {React} react element
 */
const CreatedRecipes = props => (
  <RecipesDisplay
    title="Created recipes"
    getRecipes={props.getCreatedRecipes}
    recipes={props.createdRecipes}
    {...props}
  />
);

CreatedRecipes.propTypes = {
  getCreatedRecipes: PropTypes.func.isRequired,
  createdRecipes: PropTypes.arrayOf(PropTypes.shape).isRequired
};

const mapStateToProps = state => ({
  createdRecipes: state.recipes,
});

export { CreatedRecipes as PureCreatedRecipes };
export default connect(mapStateToProps, {
  getCreatedRecipes
})(CreatedRecipes);
