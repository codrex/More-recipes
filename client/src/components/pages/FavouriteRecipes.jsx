import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getFavouriteRecipes } from '../../actions/recipeActions';
import RecipesDisplay from './RecipesDisplay';

/**
 * Carousel component
 * @param {object} props
 * @return {React} react element
 */
const FavouriteRecipes = props => (
  <RecipesDisplay
    title="Favourite recipes"
    getRecipes={props.getFavouriteRecipes}
    recipes={props.favouriteRecipes}
    {...props}
  />
);
FavouriteRecipes.propTypes = {
  getFavouriteRecipes: PropTypes.func.isRequired,
  favouriteRecipes: PropTypes.arrayOf(PropTypes.shape).isRequired,
};

const mapStateToProps = state => ({
  favouriteRecipes: state.favoriteRecipes
});
export { FavouriteRecipes as PureFavoriteRecipes };
export default connect(mapStateToProps, {
  getFavouriteRecipes
})(FavouriteRecipes);
