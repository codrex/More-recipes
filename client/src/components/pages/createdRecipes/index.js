import CreatedRecipes from './createdRecipes';

export default CreatedRecipes;
