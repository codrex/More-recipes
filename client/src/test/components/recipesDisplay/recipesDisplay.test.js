/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import store from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import { PureRecipesDisplay } from '../../../components/pages/RecipesDisplay';

const props = {
  toggleFav: jest.fn(),
  resetPageCount: jest.fn(),
  getRecipes: jest.fn(),
  resetSuccess: jest.fn(),
  recipes: [{
    name: 'cappuccino',
    category: 'drinks',
    views: 1,
    upVotes: 1,
    downVotes: 1,
    id: 1,
    image: {},
  }],
  user: {
    id: 1,
    favRecipes: [{
      id: 1,
    }],
  },
  loading: false,
  history: { push: jest.fn() },
  store: { store: store() },
};

describe('Recipes Display page component ', () => {
  test('expected to match snapshot ', () => {
    const wrapper = mount(<PureRecipesDisplay {...props} />);
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });

  test('should render NotFound component when recipes equal []', () => {
    const wrapper = mount(
      <PureRecipesDisplay {...{ ...props, recipes: [], loading: false }} />);
    const tree = toJson(wrapper);
    const notFound = wrapper.find('NotFound').length;

    expect(notFound).toBe(1);
    expect(tree).toMatchSnapshot();
    expect(tree).toBeInstanceOf(Object);
  });

  test('should render a Loader component when props.loading is true ',
    () => {
      const wrapper = mount(
        <PureRecipesDisplay {...{ ...props, loading: true }} />);
      const tree = toJson(wrapper);
      const loader = wrapper.find('div.loader').length;
      expect(loader).toBe(1);
      expect(tree).toMatchSnapshot();
      expect(tree).toBeInstanceOf(Object);
    });

  test('should call prop.toggleFav function when favourite icon is clicked',
    () => {
      const wrapper = mount(
        <PureRecipesDisplay {...{ ...props, pageCount: 2 }} />);
      const tree = toJson(wrapper);
      wrapper.find('Icon').find('.fa-heart').simulate('click');
      expect(props.toggleFav).toBeCalled();
      expect(tree).toMatchSnapshot();
    });

  test('should call resetSuccess function when component un-mounts',
    () => {
      const wrapper = mount(
        <PureRecipesDisplay {...props} />);
      const tree = toJson(wrapper);
      wrapper.unmount();
      expect(props.resetSuccess).toBeCalled();
      expect(tree).toMatchSnapshot();
    });

  test('should call setState when component receive recipes as props',
    () => {
      const wrapper = shallow(
        <PureRecipesDisplay {...props} />);
      const tree = toJson(wrapper);
      wrapper.setProps({ recipes: [{}] });
      expect(wrapper.state().recipes).toMatchObject([{}]);
      expect(tree).toMatchSnapshot();
    });
});
