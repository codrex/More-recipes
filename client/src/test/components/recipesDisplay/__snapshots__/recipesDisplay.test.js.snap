// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipes Display page component  expected to match snapshot  1`] = `
<RecipesDisplay
  getRecipes={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loading={false}
  pageCount={0}
  recipes={
    Array [
      Object {
        "category": "drinks",
        "downVotes": 1,
        "id": 1,
        "image": Object {},
        "name": "capachino",
        "upVotes": 1,
        "views": 1,
      },
    ]
  }
  resetPageCount={[Function]}
  resetSuccess={[Function]}
  store={
    Object {
      "store": [Function],
    }
  }
  title=""
  toggleFav={[Function]}
  user={
    Object {
      "favRecipes": Array [
        Object {
          "id": 1,
        },
      ],
      "id": 1,
    }
  }
>
  <div
    className="container-fluid no-padding"
    id="recipes"
  >
    <HeroArea
      image="http://res.cloudinary.com/resycom/image/upload/c_scale,q_53,w_1073/v1515038728/photo-1510693206972-df098062cb71_gxgxnb.jpg"
      title=""
    >
      <div
        className="hero-half"
      >
        <div
          className="backdrop"
        >
          <img
            alt="heroimage"
            src="http://res.cloudinary.com/resycom/image/upload/c_scale,q_53,w_1073/v1515038728/photo-1510693206972-df098062cb71_gxgxnb.jpg"
          />
          <h1
            className="display-4
          text-capitalize
          col-11
          bold
          text-center
          "
            id="display-1"
          />
        </div>
      </div>
    </HeroArea>
    <div
      className="row recipes-board d-flex flex-column"
    >
      <div
        className="bold
                text-dark
                d-flex
                justify-content-between"
        id="display-1"
      >
        <span
          className="lead"
        >
          Total Recipes: 1
        </span>
      </div>
      <RecipeGrid
        history={
          Object {
            "push": [Function],
          }
        }
        isUserFav={[Function]}
        loading={false}
        recipes={
          Array [
            Object {
              "category": "drinks",
              "downVotes": 1,
              "id": 1,
              "image": Object {},
              "name": "capachino",
              "upVotes": 1,
              "views": 1,
            },
          ]
        }
        toggleFav={[Function]}
      >
        <ul
          className="grid"
        >
          <li
            className="card-wrapper"
            key="1"
          >
            <RecipeCard
              category="drinks"
              downVotes={1}
              image={Object {}}
              index={0}
              isFav={[Function]}
              push={[Function]}
              recipeId={1}
              recipeName="capachino"
              toggleFav={[Function]}
              upVotes={1}
              views={1}
            >
              <Animation
                appear={true}
                name="fade"
                timeOut={600}
              >
                <CSSTransitionGroup
                  transitionAppear={true}
                  transitionAppearTimeout={600}
                  transitionEnter={true}
                  transitionEnterTimeout={600}
                  transitionLeave={true}
                  transitionLeaveTimeout={600}
                  transitionName="fade"
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    component="span"
                    transitionAppear={true}
                    transitionAppearTimeout={600}
                    transitionEnter={true}
                    transitionEnterTimeout={600}
                    transitionLeave={true}
                    transitionLeaveTimeout={600}
                    transitionName="fade"
                  >
                    <span>
                      <CSSTransitionGroupChild
                        appear={true}
                        appearTimeout={600}
                        enter={true}
                        enterTimeout={600}
                        key=".0"
                        leave={true}
                        leaveTimeout={600}
                        name="fade"
                      >
                        <div
                          className="recipe-card"
                          key=".0"
                        >
                          <div
                            className="recipe-card-body"
                          >
                            <Image
                              className="recipe-card-img"
                              cloudName="resycom"
                              crop="fill"
                              height="250"
                              publicId="recipe-card-placeholder_ta9ikp_iebsf2.jpg"
                              width="400"
                            >
                              <img
                                className="recipe-card-img"
                                height="250"
                                src="http://res.cloudinary.com/resycom/image/upload/c_fill,h_250,w_400/recipe-card-placeholder_ta9ikp_iebsf2.jpg"
                                width="400"
                              />
                            </Image>
                            <div
                              className="recipe-card-underlay"
                            >
                              <p
                                className="lead text-capitalize"
                              >
                                drinks
                              </p>
                              <Button
                                Tag="button"
                                className="recipe-card-underlay-btn"
                                dataTarget=""
                                dataToggle=""
                                handleClick={[Function]}
                                href=""
                                id="view-details-btn-0"
                                text="view details"
                                type="button"
                              >
                                <button
                                  className="btn recipe-card-underlay-btn"
                                  data-dismiss="modal"
                                  data-target=""
                                  data-toggle=""
                                  href=""
                                  id="view-details-btn-0"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  view details
                                </button>
                              </Button>
                            </div>
                          </div>
                          <div
                            className="recipe-card-footer"
                          >
                            <h4
                              className="ellipsis"
                            >
                              capachino
                            </h4>
                            <div
                              className="d-flex col-12 no-padding justify-content-between"
                            >
                              <div
                                className="d-flex col-8 no-padding"
                              >
                                <Icon
                                  active=""
                                  className=""
                                  dataTarget=""
                                  dataToggle=""
                                  handleClick={[Function]}
                                  iconClass="fa fa-thumbs-up recipe-card-icon "
                                  onMouseEnter={[Function]}
                                  parentClass=""
                                  pointer={false}
                                  tip="Total Upvotes: 1"
                                  toolTip={true}
                                  toolTipId="1upvote"
                                  toolTipPlace="bottom"
                                  toolTipType="dark"
                                >
                                  <div
                                    className="icon-wrapper"
                                    onClick={[Function]}
                                    onMouseEnter={[Function]}
                                    role="button"
                                    tabIndex="0"
                                  >
                                    <i
                                      className="icon fa fa-thumbs-up recipe-card-icon "
                                      data-for="1upvote"
                                      data-target=""
                                      data-tip="Total Upvotes: 1"
                                      data-toggle=""
                                    >
                                      <span>
                                        1
                                      </span>
                                    </i>
                                    <ReactTooltip
                                      effect="solid"
                                      id="1upvote"
                                      insecure={true}
                                      place="bottom"
                                      resizeHide={true}
                                      type="dark"
                                      wrapper="div"
                                    >
                                      <div
                                        className="__react_component_tooltip place-top type-dark "
                                        data-id="tooltip"
                                      />
                                    </ReactTooltip>
                                  </div>
                                </Icon>
                                <Icon
                                  active=""
                                  className=""
                                  dataTarget=""
                                  dataToggle=""
                                  handleClick={[Function]}
                                  iconClass="fa fa-thumbs-down recipe-card-icon "
                                  onMouseEnter={[Function]}
                                  parentClass=""
                                  pointer={false}
                                  tip="Total Downvotes: 1"
                                  toolTip={true}
                                  toolTipId="1downvote"
                                  toolTipPlace="bottom"
                                  toolTipType="dark"
                                >
                                  <div
                                    className="icon-wrapper"
                                    onClick={[Function]}
                                    onMouseEnter={[Function]}
                                    role="button"
                                    tabIndex="0"
                                  >
                                    <i
                                      className="icon fa fa-thumbs-down recipe-card-icon "
                                      data-for="1downvote"
                                      data-target=""
                                      data-tip="Total Downvotes: 1"
                                      data-toggle=""
                                    >
                                      <span>
                                        1
                                      </span>
                                    </i>
                                    <ReactTooltip
                                      effect="solid"
                                      id="1downvote"
                                      insecure={true}
                                      place="bottom"
                                      resizeHide={true}
                                      type="dark"
                                      wrapper="div"
                                    >
                                      <div
                                        className="__react_component_tooltip place-top type-dark "
                                        data-id="tooltip"
                                      />
                                    </ReactTooltip>
                                  </div>
                                </Icon>
                                <Icon
                                  active=""
                                  className=""
                                  dataTarget=""
                                  dataToggle=""
                                  handleClick={[Function]}
                                  iconClass="fa fa-eye recipe-card-icon"
                                  onMouseEnter={[Function]}
                                  parentClass=""
                                  pointer={false}
                                  tip="Total Number of Views: 1"
                                  toolTip={true}
                                  toolTipId="1views"
                                  toolTipPlace="bottom"
                                  toolTipType="dark"
                                >
                                  <div
                                    className="icon-wrapper"
                                    onClick={[Function]}
                                    onMouseEnter={[Function]}
                                    role="button"
                                    tabIndex="0"
                                  >
                                    <i
                                      className="icon fa fa-eye recipe-card-icon"
                                      data-for="1views"
                                      data-target=""
                                      data-tip="Total Number of Views: 1"
                                      data-toggle=""
                                    >
                                      <span>
                                        1
                                      </span>
                                    </i>
                                    <ReactTooltip
                                      effect="solid"
                                      id="1views"
                                      insecure={true}
                                      place="bottom"
                                      resizeHide={true}
                                      type="dark"
                                      wrapper="div"
                                    >
                                      <div
                                        className="__react_component_tooltip place-top type-dark "
                                        data-id="tooltip"
                                      />
                                    </ReactTooltip>
                                  </div>
                                </Icon>
                              </div>
                              <Icon
                                active=""
                                className=""
                                dataTarget=""
                                dataToggle=""
                                handleClick={[Function]}
                                iconClass="fa recipe-card-icon fav-color fav fa-heart"
                                onMouseEnter={[Function]}
                                parentClass=""
                                pointer={true}
                                tip="Unfavourite This Recipe"
                                toolTip={true}
                                toolTipId="1fav"
                                toolTipPlace="bottom"
                                toolTipType="dark"
                              >
                                <div
                                  className="icon-wrapper pointer"
                                  onClick={[Function]}
                                  onMouseEnter={[Function]}
                                  role="button"
                                  tabIndex="0"
                                >
                                  <i
                                    className="icon fa recipe-card-icon fav-color fav fa-heart"
                                    data-for="1fav"
                                    data-target=""
                                    data-tip="Unfavourite This Recipe"
                                    data-toggle=""
                                  >
                                    <span />
                                  </i>
                                  <ReactTooltip
                                    effect="solid"
                                    id="1fav"
                                    insecure={true}
                                    place="bottom"
                                    resizeHide={true}
                                    type="dark"
                                    wrapper="div"
                                  >
                                    <div
                                      className="__react_component_tooltip place-top type-dark "
                                      data-id="tooltip"
                                    />
                                  </ReactTooltip>
                                </div>
                              </Icon>
                            </div>
                          </div>
                        </div>
                      </CSSTransitionGroupChild>
                    </span>
                  </TransitionGroup>
                </CSSTransitionGroup>
              </Animation>
            </RecipeCard>
          </li>
        </ul>
      </RecipeGrid>
    </div>
  </div>
</RecipesDisplay>
`;

exports[`Recipes Display page component  should match snapshot when loading is true 1`] = `
<RecipesDisplay
  getRecipes={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loading={true}
  pageCount={0}
  recipes={
    Array [
      Object {
        "category": "drinks",
        "downVotes": 1,
        "id": 1,
        "image": Object {},
        "name": "capachino",
        "upVotes": 1,
        "views": 1,
      },
    ]
  }
  resetPageCount={[Function]}
  resetSuccess={[Function]}
  store={
    Object {
      "store": [Function],
    }
  }
  title=""
  toggleFav={[Function]}
  user={
    Object {
      "favRecipes": Array [
        Object {
          "id": 1,
        },
      ],
      "id": 1,
    }
  }
>
  <div
    className="container-fluid no-padding"
    id="recipes"
  >
    <HeroArea
      image="http://res.cloudinary.com/resycom/image/upload/c_scale,q_53,w_1073/v1515038728/photo-1510693206972-df098062cb71_gxgxnb.jpg"
      title=""
    >
      <div
        className="hero-half"
      >
        <div
          className="backdrop"
        >
          <img
            alt="heroimage"
            src="http://res.cloudinary.com/resycom/image/upload/c_scale,q_53,w_1073/v1515038728/photo-1510693206972-df098062cb71_gxgxnb.jpg"
          />
          <h1
            className="display-4
          text-capitalize
          col-11
          bold
          text-center
          "
            id="display-1"
          />
        </div>
      </div>
    </HeroArea>
    <div
      className="row recipes-board d-flex flex-column"
    >
      <RecipeGrid
        history={
          Object {
            "push": [Function],
          }
        }
        isUserFav={[Function]}
        loading={true}
        recipes={
          Array [
            Object {
              "category": "drinks",
              "downVotes": 1,
              "id": 1,
              "image": Object {},
              "name": "capachino",
              "upVotes": 1,
              "views": 1,
            },
          ]
        }
        toggleFav={[Function]}
      >
        <Loader
          loading={true}
        >
          <div
            className="loader"
          >
            <onlyUpdateForKeys(Loader)
              color="#7B4E28"
              loading={true}
              size={60}
            >
              <Loader
                color="#7B4E28"
                loading={true}
                size={60}
              >
                <div
                  className="css-igcz5d"
                />
              </Loader>
            </onlyUpdateForKeys(Loader)>
          </div>
        </Loader>
      </RecipeGrid>
    </div>
  </div>
</RecipesDisplay>
`;

exports[`Recipes Display page component  should match snapshot when recipes has a length of 0 1`] = `
<RecipesDisplay
  getRecipes={[Function]}
  history={
    Object {
      "push": [Function],
    }
  }
  loading={true}
  pageCount={0}
  recipes={Array []}
  resetPageCount={[Function]}
  resetSuccess={[Function]}
  store={
    Object {
      "store": [Function],
    }
  }
  title=""
  toggleFav={[Function]}
  user={
    Object {
      "favRecipes": Array [
        Object {
          "id": 1,
        },
      ],
      "id": 1,
    }
  }
>
  <div
    className="container-fluid no-padding"
    id="recipes"
  >
    <HeroArea
      image="http://res.cloudinary.com/resycom/image/upload/c_scale,q_53,w_1073/v1515038728/photo-1510693206972-df098062cb71_gxgxnb.jpg"
      title=""
    >
      <div
        className="hero-half"
      >
        <div
          className="backdrop"
        >
          <img
            alt="heroimage"
            src="http://res.cloudinary.com/resycom/image/upload/c_scale,q_53,w_1073/v1515038728/photo-1510693206972-df098062cb71_gxgxnb.jpg"
          />
          <h1
            className="display-4
          text-capitalize
          col-11
          bold
          text-center
          "
            id="display-1"
          />
        </div>
      </div>
    </HeroArea>
    <div
      className="row recipes-board d-flex flex-column"
    >
      <RecipeGrid
        history={
          Object {
            "push": [Function],
          }
        }
        isUserFav={[Function]}
        loading={true}
        recipes={Array []}
        toggleFav={[Function]}
      >
        <Loader
          loading={true}
        >
          <div
            className="loader"
          >
            <onlyUpdateForKeys(Loader)
              color="#7B4E28"
              loading={true}
              size={60}
            >
              <Loader
                color="#7B4E28"
                loading={true}
                size={60}
              >
                <div
                  className="css-igcz5d"
                />
              </Loader>
            </onlyUpdateForKeys(Loader)>
          </div>
        </Loader>
      </RecipeGrid>
    </div>
  </div>
</RecipesDisplay>
`;
